### PR TITLE
Sticky side navigation

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Adds sticky positioning to side navigation when page is scrolled.
When side navigation is longer than screen size it's separately scrolled.

## QA

- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/ or [demo](https://vanilla-framework-canonical-web-and-design-pr-2993.run.demo.haus/docs/examples/templates/maas-docs-grid)
- Check [docs example with sticky navigation](https://vanilla-framework-canonical-web-and-design-pr-2993.run.demo.haus/docs/examples/templates/maas-docs-grid)
- Make sure side nav works as expected
- Check if mobile view is not affected

## Screenshots

sticky

### CODE REVIEW

File: templates/docs/examples/templates/maas-docs-grid.html [hunk 1]
```diff
@@ -123,6 +122,78 @@
                 </li>
               </ul>
             </li>
+            <li class="p-side-navigation__item--title">
```
Comment (bartaz) [inline templates/docs/examples/templates/maas-docs-grid.html:125]: Just added more content to side nav and doc example to make both scroll.

**Comment by lyubomir-popov** [comment]
thanks @bartaz, scrolling works as expected, I like it. One possible enhancement on mobile - disable <main> scroll while the drawer is open?

**Comment by bartaz** [comment]
> disable scroll while the drawer is open?

We discussed it before. That requires JS and adding/removing style on `<body>`. And (quite likely) making sure on resize event to remove it as well - so felt to complicated to add to example JS.

**Comment by lyubomir-popov** [comment]
@bartaz ok, maybe we can do this in the actual vanilla docs then. +1'ed it.

**Pull request merged**